### PR TITLE
:pencil2: feature: add FuncMap to Slim engine

### PR DESCRIPTION
**Pull request by efectn**
This PR adds funcmap to Slim that wasn't added in the past.

Slim doesn't seem to support `map[string]interface{}` type for funcMap. So we have to change type of funcMap or assert interface to slim function. I prefered asserting not to break compatibility with other engines.

### CODE REVIEW

File: slim/slim.go [hunk 1]
```diff
@@ -166,7 +162,18 @@ func (e *Engine) Load() error {
...
 			return err
 		}
-		// tmpl.FuncMap(e.funcmap)
+
+		// Init func map
+		newFuncMap := make(slim.Funcs, len(e.funcMap))
+		for key, val := range e.funcMap {
+			slimFunc, ok := val.(slimFunc)
+			if !ok {
+				panic("slim: function must be compatible with slim.Func type. Slim does not support other types")
```
Comment (efectn) [inline slim/slim.go:171]: Should we use panic, error or just continue?

Comment (efectn) [reply slim/slim.go:171]: What do you think @gaby @ReneWerner87

Comment (ReneWerner87) [reply slim/slim.go:171]: Panic seems to be okay

**Review by ReneWerner87 — APPROVED** [review]

File: slim/slim.go (hunk 1 above)
Comment (efectn) [reply slim/slim.go:171]: > Panic seems to be okay

But this works inside of Render. Is it good idea to panic on requests?

**Pull request merged**